clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java

```diff
@@ -308,6 +312,31 @@ public Group revokeRolesFromGroup(List<String> roles, String group)
     return getMetalake().revokeRolesFromGroup(roles, group);
   }
 
+  /**
+   * Get the owner of a metadata object.
+   *
+   * @param object The metadata object
+   * @return The owner of the metadata object. If the metadata object doesn't set the owner, it will
+   *     return Optional.empty().
+   * @throws NotFoundException If the metadata object is not found.
+   */
+  public Optional<Owner> getOwner(MetadataObject object) throws NotFoundException {
```

Comment:
The API signature is not so good, `NotFoundException` and `Optional` have some semantic overlaps, developers may be confused when this will return empty, or when it will have NFE, unless they read the document, so we need to make this more clear and easy to understand.

Reply:
@jerqi I think maybe you can throw explicit exception, 
for example throws `NoSuchMetalakeException`, `NoSuchCatalogException`, `NoSuchTableException`, ... {

Reply:
I added an exception NoMetadataObjectException.